main.js

```diff
@@ -1,6 +1,56 @@
-import './style.css'
+const key = "18acb5642976765d66ed9ea326327f5c";
+const BASEURL = "https://api.themoviedb.org/3";
+const APIURL = `${BASEURL}/discover/movie?sort_by=popularity.desc&api_key=${key}`;
+const IMGPATH = "https://image.tmdb.org/t/p/w1280";
+const SEARCHAPI = `${BASEURL}/search/movie?&api_key=${key}`;
+const section = document.querySelector(".section");
+const main = document.querySelector(".main");
+const movielist = document.querySelector(".movielist");
+const form = document.querySelector(".form");
+const search = document.querySelector(".search");
+let query = "";
+const getMovies = async (query) => {
+  const url = query === "" ? APIURL : SEARCHAPI + "&query=" + query;
+  const response = await fetch(url);
+  const data = await response.json();
+  if (data.results.length > 0) {
+    showMovies(data.results.slice(0, 15));
+    section.style.display = "none";
+    movielist.style.display = "block";
+  } else {
+    section.style.display = "block";
+    movielist.style.display = "none";
+  }
+};
+getMovies(query);
 
-document.querySelector('#app').innerHTML = `
-  <h1>Hello Vite!</h1>
-  <a href="https://vitejs.dev/guide/features.html" target="_blank">Documentation</a>
-`
+const showMovies = (data) => {
+  data.forEach((result) => {
+    const imagePath =
+      result.poster_path === null
+        ? "no-image-icon-13.png"
+        : IMGPATH + result.poster_path;
+    const { poster_path, title } = result;
+    const box = document.createElement("div");
+
+    box.classList.add("card");
```

Comment:
The entire logic could be rewritten as template literals. Please refer - [template literals from mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)